gax/src/main/java/com/google/api/gax/rpc/OperationCallable.java

```diff
@@ -134,4 +134,35 @@ public ResponseT call(RequestT request) {
   public ApiFuture<Void> cancel(String operationName) {
     return cancel(operationName, null);
   }
+
+  /**
+   * Returns a new {@code OperationCallable} with an {@link ApiCallContext} that is used as a
+   * default when none is supplied in individual calls.
+   *
+   * @param defaultCallContext the default {@link ApiCallContext}.
+   */
+  public OperationCallable<RequestT, ResponseT, MetadataT> withDefaultCallContext(
+      final ApiCallContext defaultCallContext) {
+    return new OperationCallable<RequestT, ResponseT, MetadataT>() {
```

Comment:
(This is about all withDefaultCallContext methods in other classes as well).

Should this really be an internal anonymous class? Seems like an overkill here (also, even though it is only theoretical, but `withDefaultCallContext().withDefaultCallContext().withDefaultCallContext()` - the final bytecode result looks disgusting).

What about just letting the classes to know about "defaultContext" themselves by default? Then this OperationCallable abstract class constructor has `defaultContext` and `context` parameters and corresponding fields, and has a method "getContext()" which does something like `return defaultContext != null ? defaultContext.merge(context) : context`.

Reply:
Hmm I would argue that an explicitly named class is overkill (the former `EntryPoint*` classes); this seemed like a lower overhead way to do this - the logic is confined to one spot.